### PR TITLE
OTel mappings: avoid metrics to be rejected when  attributes are malformed

### PR DESCRIPTION
**Pull request by felixbarny**
@tetianakravchenko has noticed ingestion failures with the OTel demo. The issue is that for some reason, the `k8s.pod.ip` field is set to an empty string. While this sounds like a bug, we should handle those situations more gracefully.

### CODE REVIEW

File: x-pack/plugin/otel-data/src/main/resources/index-templates/metrics-otel@template.yaml [hunk 1]
```diff
@@ -27,14 +27,3 @@ template:
...
-      - ecs_ip:
-          mapping:
-            type: ip
-          path_match: [ "ip", "*.ip", "*_ip" ]
-          match_mapping_type: string
-      - all_strings_to_keywords:
-          mapping:
-            ignore_above: 1024
-            type: keyword
-          match_mapping_type: string
```
Comment (axw) [inline x-pack/plugin/otel-data/src/main/resources/index-templates/metrics-otel@template.yaml:40]: Why are these removed?

**Review by axw — COMMENTED** [review]
We'll need to bump the template version too

File: x-pack/plugin/otel-data/src/main/resources/index-templates/metrics-otel@template.yaml (hunk 1 above)
Comment (felixbarny) [reply x-pack/plugin/otel-data/src/main/resources/index-templates/metrics-otel@template.yaml:40]: This is a leftover from https://github.com/elastic/elasticsearch/pull/114042. The content has moved to [`ecs-tsdb@mappings`](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/otel-data/src/main/resources/component-templates/ecs-tsdb%40mappings.yaml).

**Review by axw — APPROVED** [review]

**Comment by felixbarny** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/114878)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

File: x-pack/plugin/otel-data/src/main/resources/component-templates/metrics-otel@mappings.yaml [hunk 2]
```diff
@@ -3,6 +3,8 @@ _meta:
   description: Default mappings for the OpenTelemetry metrics index template installed by x-pack
   managed: true
 template:
+  settings:
+    index.mapping.ignore_malformed: true
```
Comment (gregkalapos) [inline x-pack/plugin/otel-data/src/main/resources/component-templates/metrics-otel@mappings.yaml:7]: Late on this PR, but this is just a nit, so wouldn't change anything.

For traces and logs we set this implicitly via `index.mode: logsdb`, right? I wonder, maybe we should add this also to `logs-otel@mappings.yaml` and `traces-otel@mappings.yaml` explicitly. It would be better for readability - with OTel, we aim to never reject data, so I think it'd be worth the extra line, even if the current default is the same. Or maybe even move this to one of the central `yaml` files used by all signals.

Comment (felixbarny) [reply x-pack/plugin/otel-data/src/main/resources/component-templates/metrics-otel@mappings.yaml:7]: Right, for the other signals, it's implicit due to LogsDB. Good idea to move it to a central place used by all signals.

Comment (gregkalapos) [reply x-pack/plugin/otel-data/src/main/resources/component-templates/metrics-otel@mappings.yaml:7]: Added to https://github.com/elastic/elasticsearch/issues/111877
